Bottom: ca6c7814600f3cd5e3dd1e8cd44cfac895959b53
Top:    8f0704dcc42b473e624ab49402458f7bae519904
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-01-07 01:09:17 +0000

Fixup code to determine whether we draw pins/vias on a given layer

The net change of this patch is that we consider whether to draw pins/vias
for each layer within the layer group. Previously, if we encoundered a layer
named "outline" or "route" within the layer group, all _subsequent_ layers
within that group would not have their pins/vias rendered.

As ordering of the layers within the group should not be important like this,
change behaviour to apply test on a per-layer basis.


---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 0888e88..71b60df 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -1358,7 +1358,7 @@ GhidDrawMask (int side, BoxType * screen)
 static int
 GhidDrawLayerGroup (int group, const BoxType * screen)
 {
-  int i, rv = 1;
+  int i;
   int layernum;
   int side;
   struct poly_info info;
@@ -1376,10 +1376,10 @@ GhidDrawLayerGroup (int group, const BoxType * screen)
   for (i = n_entries - 1; i >= 0; i--) {
     layernum = layers[i];
     Layer = PCB->Data->Layer + layers[i];
+    bool is_outline;
 
-    if (strcmp (Layer->Name, "outline") == 0 ||
-        strcmp (Layer->Name, "route") == 0)
-      rv = 0;
+    is_outline = strcmp (Layer->Name, "outline") == 0 ||
+                 strcmp (Layer->Name, "route") == 0;
 
     if (layernum < max_copper_layer && Layer->On) {
 
@@ -1388,7 +1388,7 @@ GhidDrawLayerGroup (int group, const BoxType * screen)
 
       first_run = 0;
 
-      if (rv && !TEST_FLAG (THINDRAWFLAG, PCB)) {
+      if (!is_outline && !TEST_FLAG (THINDRAWFLAG, PCB)) {
         /* Mask out drilled holes on this layer */
         hidgl_flush_triangles (&buffer);
         glPushAttrib (GL_COLOR_BUFFER_BIT);
@@ -1411,7 +1411,7 @@ GhidDrawLayerGroup (int group, const BoxType * screen)
         gui->end_layer ();
         gui->set_layer (0, group, 0);
 
-        if (rv && !TEST_FLAG (THINDRAWFLAG, PCB)) {
+        if (!is_outline && !TEST_FLAG (THINDRAWFLAG, PCB)) {
           hidgl_flush_triangles (&buffer);
           glPushAttrib (GL_COLOR_BUFFER_BIT);
           glColorMask (0, 0, 0, 0);
@@ -1424,7 +1424,7 @@ GhidDrawLayerGroup (int group, const BoxType * screen)
       }
 
       /* Draw pins, vias and pads on this layer */
-      if (!global_view_2d && rv) {
+      if (!global_view_2d && !is_outline) {
         if (PCB->PinOn) r_search (PCB->Data->pin_tree, screen, NULL, pin_inlayer_callback, Layer);
         if (PCB->ViaOn) r_search (PCB->Data->via_tree, screen, NULL, via_inlayer_callback, Layer);
         if (PCB->PinOn && group == top_group)
